grpc-gcp/src/main/java/com/google/grpc/gcp/GcpManagedChannel.java

```diff
@@ -76,10 +79,16 @@
   @GuardedBy("bindLock")
   final Map<String, ChannelRef> affinityKeyToChannelRef = new HashMap<String, ChannelRef>();
 
+  // Map from a broken channel id to the remapped affinity keys (key => ready channel id).
+  private final ConcurrentHashMap<Integer, ConcurrentHashMap<String, Integer>> fallbackMap =
+      new ConcurrentHashMap<>();
+
   @VisibleForTesting
   @GuardedBy("this")
   final List<ChannelRef> channelRefs = new ArrayList<ChannelRef>();
 
+  private final HashMap<Integer, ChannelRef> channelRefById = new HashMap<>();
```

Comment:
Yes, I want to look into what can be done with the `channelRefs` But let me do that in a separate PR, because I have another PR coming that also touches these parts.